contrib/IECoreUSD/src/IECoreUSD/USDScene.cpp

```diff
@@ -0,0 +1,832 @@
+	pxr::UsdGeomBBoxCache bboxCache( m_root->getTime( time ), purposes );
+
+	pxr::GfBBox3d bbox = bboxCache.ComputeLocalBound( m_prim );
+
+	return convert( bbox );
+}
+
+ConstDataPtr USDScene::readTransform( double time ) const
+{
+	return new IECore::M44dData( readTransformAsMatrix( time ) );
+}
+
+Imath::M44d USDScene::readTransformAsMatrix( double time ) const
+{
+	pxr::UsdGeomXformable transformable( m_prim );
+	pxr::GfMatrix4d transform;
+	bool reset = false;
+
+	transformable.GetLocalTransformation( &transform, &reset, m_root->getTime( time ) );
+	return convert( transform );
+}
+
+ConstObjectPtr USDScene::readAttribute( const SceneInterface::Name &name, double time ) const
+{
+	return nullptr;
+}
+
+ConstObjectPtr USDScene::readObject( double time ) const
+{
+	return convert( m_prim, m_root->getTime( time ) );
+}
+
+SceneInterface::Name USDScene::name() const
+{
+	return SceneInterface::Name( m_prim.GetName() );
+}
+
+void USDScene::path( SceneInterface::Path &p ) const
+{
+	std::vector<std::string> parts;
+	pxr::SdfPath path = m_prim.GetPath();
+	boost::split( parts, path.GetString(), boost::is_any_of( "/" ) );
+
+	p.reserve( parts.size() );
+
+	for( auto part : parts )
+	{
+		if( part != "" )
+		{
+			p.push_back( IECore::InternedString( part ) );
+		}
+	}
+}
+
+bool USDScene::hasBound() const
+{
+	return true;
+}
+
+void USDScene::writeBound( const Imath::Box3d &bound, double time )
+{
+	throw IECore::NotImplementedException("USDScene::writeBound not supported" );
+}
+
+void USDScene::writeTransform( const Data *transform, double time )
+{
+	throw IECore::NotImplementedException("USDScene::writeTransform not supported" );
+}
+
+bool USDScene::hasAttribute( const SceneInterface::Name &name ) const
+{
+	return false;
+}
+
+void USDScene::attributeNames( SceneInterface::NameList &attrs ) const
+{
+
+}
+
+void USDScene::writeAttribute( const SceneInterface::Name &name, const Object *attribute, double time )
+{
+	throw IECore::NotImplementedException("USDScene::writeAttribute not supported" );
+}
+
+bool USDScene::hasTag( const SceneInterface::Name &name, int filter ) const
+{
+	return false;
+}
+
+void USDScene::readTags( SceneInterface::NameList &tags, int filter ) const
+{
+
+}
+
+void USDScene::writeTags( const SceneInterface::NameList &tags )
+{
+	throw IECore::NotImplementedException("USDScene::writeTags not supported" );
+}
+
+bool USDScene::hasObject() const
+{
+	return isConvertable( m_prim );
+}
+
+PrimitiveVariableMap USDScene::readObjectPrimitiveVariables( const std::vector<InternedString> &primVarNames, double time ) const
+{
+	return PrimitiveVariableMap();
+}
+
+void USDScene::writeObject( const Object *object, double time )
+{
+	throw IECore::NotImplementedException("USDScene::writeObject not supported" );
+}
+
+bool USDScene::hasChild( const SceneInterface::Name &name ) const
+{
+	SceneInterface::NameList children;
+	childNames( children );
+	for( const auto &i : children )
```

Comment:
Is linear search really the best we can do? Does `USDPrim::GetChild()` not work?